atspi-common/src/events/mod.rs

```diff
@@ -936,24 +813,229 @@ pub trait BusProperties {
 	/// A registry event string for registering for event receiving via the `RegistryProxy`.
 	/// This should be deprecated in favour of composing the string from [`Self::DBUS_MEMBER`] and [`Self::DBUS_INTERFACE`].
 	const REGISTRY_EVENT_STRING: &'static str;
+}
 
+#[cfg(feature = "zbus")]
+pub trait MessageConversion: BusProperties {
 	/// What is the body type of this event.
 	type Body: Type + Serialize + for<'a> Deserialize<'a>;
 
-	/// Build the event from the object pair (`ObjectRef` and the Body).
+	/// Build an event from a [`zbus::Message`] reference.
+	/// This function will not check for any of the following error conditions:
+	///
+	/// - That the message has an interface: [`type@AtspiError::MissingInterface`]
+	/// - That the message interface matches the one for the event: [`type@AtspiError::InterfaceMatch`]
+	/// - That the message has an member: [`type@AtspiError::MissingMember`]
+	/// - That the message member matches the one for the event: [`type@AtspiError::MemberMatch`]
+	/// - That the message has an signature: [`type@AtspiError::MissingSignature`]
+	/// - That the message signature matches the one for the event: [`type@AtspiError::SignatureMatch`]
+	///
+	/// Therefore, this should only be used when one has checked the above conditions.
+	/// These must be checked manually.
+	/// Alternatively, there is the [`MessageConversionExt::try_from_message`] that will check these
+	/// conditions for you.
+	///
+	/// This type also implements `TryFrom<&zbus::Message>`; consider using this if you are not an
+	/// internal developer.
+	///
+	/// # Errors
+	///
+	/// It is possible to get a [`type@AtspiError::Zvariant`] error if you do not check the proper
+	/// conditions before calling this.
+	fn try_from_validated_message(msg: &zbus::Message) -> Result<Self, AtspiError>
```

Comment:
My apologies if we addressed this before but as the qualifier "unchecked" is more common in the ecosystem, shouldn't we call this `try_from_message_unchecked` to raise the proper alarm bells?
Semantically `try_from_validated_message` may be more fitting but having `unchecked` in there would make users uncomfortable precisely the right amount. What do you think?

Reply:
Fixed. `try_from_validated_message -> from_message_unchecked` throughout the codebase.